explain-joins.md

```diff
@@ -0,0 +1,219 @@
+
+If the estimated rows that need to be joined is small (typically less than 10000 rows), using the Index Join method is preferable. This method of join works similar to the primary method of join used in MySQL. In the following example, the operator `IndexReader_31(Build)` first reads the table `t1`. For each row that matches, TiDB will probe the table `t2`:
+
+{{< copyable "sql" >}}
+
+```sql
+EXPLAIN SELECT /*+ INL_JOIN(t1, t2) */ * FROM t1 INNER JOIN t2 ON t1.id = t2.t1_id;
+```
+
+```sql
++---------------------------------+-----------+-----------+------------------------------+--------------------------------------------------------------------------------+
+| id                              | estRows   | task      | access object                | operator info                                                                  |
++---------------------------------+-----------+-----------+------------------------------+--------------------------------------------------------------------------------+
+| IndexJoin_10                    | 180000.00 | root      |                              | inner join, inner:IndexLookUp_9, outer key:test.t1.id, inner key:test.t2.t1_id |
+| ├─TableReader_28(Build)         | 142020.00 | root      |                              | data:TableFullScan_27                                                          |
+| │ └─TableFullScan_27            | 142020.00 | cop[tikv] | table:t1                     | keep order:false                                                               |
+| └─IndexLookUp_9(Probe)          | 1.27      | root      |                              |                                                                                |
+|   ├─IndexRangeScan_7(Build)     | 1.27      | cop[tikv] | table:t2, index:t1_id(t1_id) | range: decided by [eq(test.t2.t1_id, test.t1.id)], keep order:false            |
+|   └─TableRowIDScan_8(Probe)     | 1.27      | cop[tikv] | table:t2                     | keep order:false                                                               |
++---------------------------------+-----------+-----------+------------------------------+--------------------------------------------------------------------------------+
+6 rows in set (0.00 sec)
+```
+
+Index join is efficient in memory usage, but may be slower to execute than other join methods when a large number of probe operations are required. Consider also the following query:
+
+```sql
+SELECT * FROM t1 INNER JOIN t2 ON t1.id=t2.t1_id WHERE t1.pad1 = 'value' and t2.pad1='value';
+```
+
+On an inner join, TiDB will apply join reordering and might access either `t1` or `t2` first. Assuming that TiDB selects `t1` as the first table to apply the `Build` step, it will be able to filter on the predicate `t1.col = 'value'` before probing the table `t2`. The filter for the predicate `t2.col='value'` will be applied on each probe of table `t2`, which may be less efficient than other join methods.
```

Comment:
Could we provide a query example showing how we tune Hash Join into Index Join by adding a suitable index for better performance?